chains/chains_test.go

```diff
@@ -38,3 +41,24 @@ func TestApply(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, numInputs, len(results), "number of inputs and results not equal")
 }
+
+func TestApplyWithCanceledContext(t *testing.T) {
```

Comment:
We should uplift the `TestApply` function to be more comprehensive - it should test that an expected output exists for a given input.